resilience4j-retry/src/main/java/io/github/resilience4j/retry/internal/RetryImpl.java

```diff
@@ -359,19 +357,20 @@ private long handleOnError(Throwable throwable) {
         public long onResult(T result) {
             if (null != resultPredicate && resultPredicate.test(result)) {
                 totalAttemptsCounter.increment();
-                int attempt = numOfAttempts.incrementAndGet();
-                if (attempt >= maxAttempts) {
-                    if(consumeResultBeforeRetryAttempt != null){
-                        consumeResultBeforeRetryAttempt.accept(attempt, result);
-                    }
```

Comment:
(FYI) On `AsyncContextImpl`, `consumeResultBeforeRetryAttempt ` is only invoked when `attempt > maxAttempts` and it seems bug!

On `ContextImplt`,  `consumeResultBeforeRetryAttempt ` is invoked on each retry well.